lib/ansible/modules/cloud/amazon/iam_server_certificate_info.py

```diff
@@ -36,136 +42,166 @@
-            "server_certificate_id": "ADWAJXWTZAXIPIMQHMJPO",
-            "certificate_body": "-----BEGIN CERTIFICATE-----\nbunch of random data\n-----END CERTIFICATE-----",
-            "server_certificate_name": "server-cert-name",
-            "expiration": "2017-06-15T12:00:00+00:00",
-            "path": "/",
-            "arn": "arn:aws:iam::911277865346:server-certificate/server-cert-name"
-        }
+
+def _get_server_certs(connection, module, name, path):
     """
-    results = dict()
-    try:
-        if name:
-            server_certs = [iam.get_server_certificate(ServerCertificateName=name)['ServerCertificate']]
-        else:
-            server_certs = iam.list_server_certificates()['ServerCertificateMetadataList']
+    Get one or more IAM server certificates facts based on name or path. If not found, return None.
 
+    :param connection: AWS boto3 glue connection
+    :param module: Ansible module
+    :param name: The name of the server certificate
+    :param path: The path of the server certificate
+    :return: IAM server certificates dict or None if not found
+    """
+
+    response = dict()
+    response['certificates'] = []
+
+    if name:
+        try:
+            response['certificates'].append(connection.get_server_certificate(ServerCertificateName=name)['ServerCertificate'])
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                return response['certificates']
+            else:
+                module.fail_json_aws(e, msg="The specified server certificates could not be found ")
+        except botocore.exceptions.BotoCoreError as e:
+            module.fail_json_aws(e, msg="The specified server certificates could not be found ")
+    else:
+        server_certs = list_iam_server_certificates_with_backoff(connection, path)['ServerCertificateMetadataList']
+
+    if not name:
         for server_cert in server_certs:
-            if not name:
-                server_cert = iam.get_server_certificate(ServerCertificateName=server_cert['ServerCertificateName'])['ServerCertificate']
-            cert_md = server_cert['ServerCertificateMetadata']
-            results[cert_md['ServerCertificateName']] = {
-                'certificate_body': server_cert['CertificateBody'],
-                'server_certificate_id': cert_md['ServerCertificateId'],
-                'server_certificate_name': cert_md['ServerCertificateName'],
-                'arn': cert_md['Arn'],
-                'path': cert_md['Path'],
-                'expiration': cert_md['Expiration'].isoformat(),
-                'upload_date': cert_md['UploadDate'].isoformat(),
-            }
+            try:
+                response['certificates'].append(
+                    connection.get_server_certificate(ServerCertificateName=server_cert['ServerCertificateName'])['ServerCertificate'])
+            except botocore.exceptions.ClientError as e:
```

Comment:
Did you mean to simply drop any other ClientErrors on the floor here?  (You're also not handling BotoCoreError this time around.)

Reply:
I have handled this now. Thank you for pointing this out